src/NuGet.Core/NuGet.Frameworks/FrameworkNameHelpers.cs

```diff
@@ -38,23 +38,23 @@ public static string GetVersionString(Version version)
                 }
             }
 
-            return versionString;
+            return versionString!;
         }
 
-        public static Version GetVersion(string versionString)
+        public static Version GetVersion(string? versionString)
         {
-            Version version = null;
+            Version version;
```

Comment:
You could also simplify this method to just return the values and reduce nesting while you're at it.  Line 50, 57, and 70 could just `return something`.